contents/tutorials/python-v6-migration.md

```diff
@@ -0,0 +1,334 @@
+| **Breaking** | Order of arguments updated for capture-related methods. You must update calls to use kwargs before upgrading |
+| **Breaking** | Capture methods now return the UUID of the captured event, rather than the full event object |
+| **Breaking** | Deletes unmaintained exception capture integrations, which is replaced by the general purpose Django middleware |
+
+<CalloutBox icon="IconInfo" title="Upgrade best practices">
+
+You should always pin versions using the [compatible release clause](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release) to avoid auto-upgrading to an incompatible version accidentally.
+
+</CalloutBox>
+
+## New feature: contexts
+
+Version `6.x` introduces contexts. Events captured under the same context scope will share properties set on the context. This can be used to identify who, where, and when an event is triggered.
+
+This is very convenient for server-side use cases like [Django](/docs/libraries/django#django-contexts-middleware) apps, where each request might be handled for a different user. 
+
+```python
+from posthog import new_context, tag, set_context_session, identify_context
+
+with new_context():
+  # Identify user and session
+  set_context_session(session_id)
+  identify_context(user_id)
+
+  # Set properties share by events in the context
+  tag("transaction_id", "abc123")
+
+  # This event is captured with the distinct ID, session ID, and tags set above
+  posthog.capture("order_processed")
+```
+
+Contexts can be passed down the call tree. You can iteratively add tags to the context to keep track of events leading up to a capture call. For example:
+
+```python
+from django.http import HttpResponse
+from posthog import new_context, tag, set_context_session, identify_context, capture
+from posthog.contexts import get_tags
+
+def handle_order(request):
+    with new_context():
+        set_context_session(session_id)
+        identify_context(user_id)
+        validate_order()
+    return HttpResponse("Hello, World!")
+
+def validate_order():
+    tag("order_steps", ["validate_order"])
+    process_payment()
+
+def process_payment():
+    tag("order_steps", [*get_tags()["order_steps"], "process_payment"])
+    update_inventory()
+
+def update_inventory():
+    tag("order_steps", [*get_tags()["order_steps"], "update_inventory"])
+<<<<<<< HEAD
+    capture("order_completed")
+```
+
+When `capture` is finally called, the `order_completed` event will have a custom event property called `order_steps` with a list of values `"validate_order"`, `"process_payment"`, and `"update_inventory"`.
+
+Contexts can also be nested. This means that:
+=======
+    capture("order_completed") # HIGHLIGHT
+```
+
+When capture is finally called, the `order_completed` event will have a custom event property called `order_steps` with the value `["validate_order", "process_payment", "update_inventory"]`.
+>>>>>>> 142fc05a2fd18a12bd6c9d3bb3f2751b47f955a2
```

Comment:
I think this all needs some fixing, but because GitHub sucks at multi-line changes, it is broken.